server/docker-base/Dockerfile

```diff
@@ -581,6 +435,82 @@ COPY template.build.gradle ${EXTENSION_BUILD_GRADLE_TEMPLATE}
+ENV XCODE_12_CLANG_VERSION  12.0.5
+ENV SWIFT_5_VERSION         5.0
+
+# The updated install flow
+RUN \
+  apt-get update && \
+  apt-get install -y git cmake && \
+  cd /tmp && \
+  TAPITMP=/usr/local/tapi1.6 && \
+  mkdir $TAPITMP && \
+  git clone https://github.com/tpoechtrager/apple-libtapi.git && \
+  cd apple-libtapi && \
+  git checkout a66284251b46d591ee4a0cb4cf561b92a0c138d8 && \
+  INSTALLPREFIX=$TAPITMP ./build.sh && \
+  ./install.sh && \
+  cd .. && \
+  git clone https://github.com/tpoechtrager/cctools-port.git && \
+  cd cctools-port/cctools && \
+  git checkout 6c438753d2252274678d3e0839270045698c159b && \
+  ./configure --prefix=/usr/local --target=arm-apple-darwin19 --with-libtapi=$TAPITMP && \
+  make -j8 && \
+  make install && \
+  make distclean && \
+  ./configure --prefix=/usr/local --target=x86_64-apple-darwin19 --with-libtapi=$TAPITMP && \
+  make -j8 && \
+  make install && \
+  make distclean && \
+  cd ../.. && \
+  rm -rf apple-libtapi && \
+  rm -rf cctools-port
+
+RUN \
+  echo "iOS 14 + macOS 10/11 + XCode 12" && \
+  wget -q -O - ${DM_PACKAGES_URL}/iPhoneOS${IOS_14_VERSION}.sdk.tar.gz | tar xz -C ${PLATFORMSDK_DIR} && \
+  mv ${PLATFORMSDK_DIR}/iPhoneOS.sdk ${PLATFORMSDK_DIR}/iPhoneOS${IOS_14_VERSION}.sdk && \
+  wget -q -O - ${DM_PACKAGES_URL}/iPhoneSimulator${IOS_14_VERSION}.sdk.tar.gz | tar xz -C ${PLATFORMSDK_DIR} && \
+  mv ${PLATFORMSDK_DIR}/iPhoneSimulator.sdk ${PLATFORMSDK_DIR}/iPhoneSimulator${IOS_14_VERSION}.sdk && \
+  wget -q -O - ${DM_PACKAGES_URL}/MacOSX${MACOS_10_15_VERSION}.sdk.tar.gz | tar xz -C ${PLATFORMSDK_DIR} && \
+  mv ${PLATFORMSDK_DIR}/MacOSX.sdk ${PLATFORMSDK_DIR}/MacOSX${MACOS_10_15_VERSION}.sdk && \
+  wget -q -O - ${DM_PACKAGES_URL}/MacOSX${MACOS_11_VERSION}.sdk.tar.gz | tar xz -C ${PLATFORMSDK_DIR} && \
+  mv ${PLATFORMSDK_DIR}/MacOSX.sdk ${PLATFORMSDK_DIR}/MacOSX${MACOS_11_VERSION}.sdk && \
+  wget -q -O - ${DM_PACKAGES_URL}/XcodeDefault${XCODE_12_VERSION}.xctoolchain.linux.tar.gz | tar xz -C ${PLATFORMSDK_DIR} && \
+  mv ${PLATFORMSDK_DIR}/XcodeDefault.xctoolchain ${PLATFORMSDK_DIR}/XcodeDefault${XCODE_12_VERSION}.xctoolchain
+
+# This part is not yet separable between xcode versions, but should work
+RUN \
+  ln -s /usr/local/bin/llvm-ar /usr/local/bin/ar && \
+  ln -s ${PLATFORMSDK_DIR}/XcodeDefault${XCODE_12_VERSION}.xctoolchain/usr/lib/arc /usr/local/lib/arc
+
+RUN mkdir -p /usr/local/lib/clang/10.0.1/lib/darwin && \
+    ln -s /opt/platformsdk/XcodeDefault${XCODE_12_VERSION}.xctoolchain/usr/lib/clang/${XCODE_12_CLANG_VERSION}/lib/darwin/libclang_rt.asan_osx_dynamic.dylib \
+    /usr/local/lib/clang/10.0.1/lib/darwin/libclang_rt.asan_osx_dynamic.dylib
+
+# For backwards compatbility, we must also keep a copy of the library folder
+# Remove when 1.2.185 is the base line
+RUN ln -s /opt/platformsdk/XcodeDefault${XCODE_12_VERSION}.xctoolchain/usr/lib/clang/${XCODE_12_CLANG_VERSION} \
+    /opt/platformsdk/XcodeDefault${XCODE_12_VERSION}.xctoolchain/usr/lib/clang/12.0.0
```

Comment:
We use the new XCode but older build.yml's refer to the 12.0.0 clang version, so we need to make a symlink.